components/squad/wikis/arenaofvalor/squad_custom.lua

```diff
@@ -0,0 +1,120 @@
+---
+-- @Liquipedia
+-- wiki=arenaofvalor
+-- page=Module:Squad/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Squad = require('Module:Squad')
+local SquadRow = require('Module:Squad/Row')
+local Json = require('Module:Json')
+local Variables = require('Module:Variables')
+local ReferenceCleaner = require('Module:ReferenceCleaner')
+local Class = require('Module:Class')
+local String = require('Module:String')
+
+local CustomSquad = {}
+
+function CustomSquad.header(self)
+	local makeHeader = function(wikiText)
+		local headerCell = mw.html.create('th')
+
+		if wikiText == nil then
+			return headerCell
+		end
+
+		return headerCell:wikitext(wikiText):addClass('divCell')
+	end
+
+	local headerRow = mw.html.create('tr'):addClass('HeaderRow')
+
+		headerRow	:node(makeHeader('ID'))
+					:node(makeHeader('Name'))
+					:node(makeHeader())
+					:node(makeHeader('Position'))
+					:node(makeHeader('Join Date'))
+	if self.type == Squad.TYPE_FORMER then
+		headerRow	:node(makeHeader('Leave Date'))
+					:node(makeHeader('New Team'))
+	elseif self.type == Squad.TYPE_INACTIVE then
+		headerRow:node(makeHeader('Inactive Date'))
+	end
+
+	self.content:node(headerRow)
+
+	return self
+end
+
+local ExtendedSquadRow = Class.new(SquadRow)
+
+function ExtendedSquadRow:position(args)
+	local cell = mw.html.create('td')
+	cell:addClass('Position')
+
+	if not String.isEmpty(args.position) then
+		cell:node(mw.html.create('div'):addClass('MobileStuff'):wikitext('Position:&nbsp;'))
+		cell:wikitext('' .. args.position .. '')
+	end
+
+	self.content:node(cell)
+
+	self.lpdbData['position'] = args.position
+
+	return self
+end
+
+function CustomSquad.run(frame)
+	local squad = Squad()
+
+	-- It looks like we cannot extend Squad because overwriting
+	-- the function makes the other functions inaccessible?
+	-- TODO: investigate
```

Comment:
Not aware of this, any context?

Reply:
unsure of this too, these notes seems to be existed since the original revisions but theres also no noticeable issues on display (no idea code-wise)

Reply:
You made me wonder if this is something I wrote myself 😆 

Thankfully, no. But yea we could use some context here

Reply:
These are the closest I can find relate to the origins of these notes
https://liquipedia.net/mobilelegends/index.php?title=Module%3ASquad%2FCustom&type=revision&diff=2855&oldid=2853

Appears to be involving two stuff about **ExtendedSquad** and **CustomSquad**

(due to I have totally 0 basic coding knowledge, my replies may look unusual comparing to other users who regularly do PRs)

Reply:
@hjpalpha Any ideas?

Reply:
sry no idea
the only custom squad implementation i touched was the sc2 one and there we do not have the issue (as we do not have the ExtendedSquadRow stuff)

Reply:
Oh my, I did write that code! I remember it now 😐